plugins/meta/portmap/portmap.go

```diff
@@ -120,10 +120,21 @@ func checkPorts(config *PortMapConf, containerNet net.IPNet) error {
 	dnatChain := genDnatChain(config.Name, config.ContainerID)
 	fillDnatRules(&dnatChain, config, containerNet)
 
-	ip4t := maybeGetIptables(false)
-	ip6t := maybeGetIptables(true)
+	ip4t, err4 := maybeGetIptables(false)
+	ip6t, err6 := maybeGetIptables(true)
 	if ip4t == nil && ip6t == nil {
-		return fmt.Errorf("neither iptables nor ip6tables usable")
+		err := fmt.Errorf("neither iptables nor ip6tables is usable")
+		if err4 != nil {
```

Comment:
If `ip4t` and `ip6t` are both nil, then `err4` and `err6` should both be non-nil logically. So IMO this judgement is not necessary.